lib/core/utils/preload-cssom.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Returns a then(able) queue of CSSStyleSheet(s)
+ * @param {Object} ownerDocument document object to be inspected for stylesheets
+ * @param {number} timeout on network request for stylesheet that need to be externally fetched
+ * @param {Function} getSheetFromTextFn a utility function to generate a style sheet from text
+ * @return {Object} queue
+ * @private
+ */
+function loadCssom(ownerDocument, timeout, getSheetFromTextFn) {
+	const q = axe.utils.queue();
+
+	Array.from(ownerDocument.styleSheets).forEach(sheet => {
+		if (sheet.disabled) {
+			return;
+		}
+
+		try {
+			// attempt to resolve if sheet is relative/ same domain
+			sheet.cssRules && q.defer(resolve => resolve(sheet));
+		} catch (e) {
+			const deferredSheet = (resolve, reject) => {
+				axe.utils
+					.xhrQ({
+						url: sheet.href,
+						timeout
+					})
+					.then(xhrResponse => {
+						if (
+							xhrResponse &&
+							Array.isArray(xhrResponse) &&
+							xhrResponse.length
+						) {
+							xhrResponse.forEach(r => {
+								const text = r.responseText ? r.responseText : r.response;
```

Comment:
In what case(s) would `.responseText` not be truthy (where `.response` is)? Should we document that here?

Reply:
Ignore the above - likely will change, once axios is in place - https://github.com/dequelabs/axe-core/pull/990